### PR TITLE
SUP-17624 Fix format plugin configuration

### PR DESCRIPTION
**Pull request by yrucrem**
https://jira-gentics.atlassian.net/browse/SUP-17624

### CODE REVIEW

File: src/plugins/common/format/lib/format-plugin.js [hunk 1]
```diff
@@ -710,10 +710,10 @@ define('format/format-plugin', [
 	/**
 	 * register the plugin with unique name
 	 */
-	return Plugin.create('format', {
-		config: DEFAULT_CONFIG,
+	var plugin = Plugin.create('format', {
+		buttonConfig: structuredClone(DEFAULT_BUTTON_CONFIG),
```
Comment (deckdom) [inline src/plugins/common/format/lib/format-plugin.js:714]: Cite/Quote, or some plugin adds some elements to `config`, must be changed to `buttonConfig`

File: src/plugins/common/format/lib/format-plugin.js [hunk 2]
```diff
@@ -726,7 +726,7 @@ define('format/format-plugin', [
 		 *
 		 * @todo new buttons needed for 'code'
 		 */
-		availableButtons: Object.keys(DEFAULT_CONFIG).concat(REMOVE_FORMAT_ID),
```
Comment (deckdom) [inline src/plugins/common/format/lib/format-plugin.js:729]: Check if another plugin modifies this as well

File: src/plugins/common/format/lib/format-plugin.js (hunk 1 above)
Comment (yrucrem) [reply src/plugins/common/format/lib/format-plugin.js:714]: Done

File: src/plugins/common/format/lib/format-plugin.js (hunk 2 above)
Comment (yrucrem) [reply src/plugins/common/format/lib/format-plugin.js:729]: No other instances found.

**Pull request merged**